client/blocks/login/two-factor-authentication/verification-code-input.jsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLink from 'components/external-link';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormButton from 'components/forms/form-button';
+import FormTextInput from 'components/forms/form-text-input';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormInputValidation from 'components/forms/form-input-validation';
+import Card from 'components/card';
+import { localize } from 'i18n-calypso';
+import { loginUserWithTwoFactorVerificationCode } from 'state/login/actions';
+import {
+	getTwoFactorAuthId,
+	getTwoFactorAuthNonce,
+} from 'state/login/selectors';
+
+class VerificationCodeInput extends Component {
+	static propTypes = {
+		rememberMe: PropTypes.bool.isRequired,
+		twoStepId: PropTypes.string.isRequired,
+		twoStepNonce: PropTypes.string.isRequired,
+	};
+
+	state = {
+		twoStepCode: '',
+		error: null
```

Comment:
When is `error` ever set to a non-null value? We don't have a `catch` handler for the `loginUserWithTwoFactorVerificationCode` promise, so it does not seem to me that this ever happens.